www/qserv/js/QservWorkerSchedulers.js

```diff
@@ -0,0 +1,160 @@
+            this._initialized = true;
+            this._prevTimestamp = 0;
+
+            let html = `
+<div class="row">
+  <div class="col">
+    <table class="table table-sm table-hover table-bordered" id="fwk-qserv-schedulers">
+      <thead class="thead-light">
+        <tr>
+          <th>worker</th>
+          <th>scheduler</th>
+          <th>priority</th>
+          <th>in-flight : #tasks</th>
+          <th>: chunk(#tasks)</th>
+          <th>queued : #tasks</th>
+          <th>: QID(#tasks)</th>
+        </tr>
+      </thead>
+      <caption class="updating">Loading...</caption>
+      <tbody></tbody>
+    </table>
+  </div>
+</div>`;
+            this.fwk_app_container.html(html);
+        }
+        
+        /// @returns JQuery table object displaying the load of schedulers on each worker
+        _table() {
+            if (this._table_obj === undefined) {
+                this._table_obj = this.fwk_app_container.find('table#fwk-qserv-schedulers');
+            }
+            return this._table_obj;
+        }
+
+        /**
+         * Load data from a web servie then render it to the application's page.
+         */
+        _load() {
+            if (this._loading === undefined) this._loading = false;
+            if (this._loading) return;
+            this._loading = true;
+
+            this._table().children('caption').addClass('updating');
+
+            Fwk.web_service_GET(
+                "/replication/v1/qserv/worker/status",
+                {'timeout_sec': 2},
+                (data) => {
+                    this._display(data.status);
+                    Fwk.setLastUpdate(this._table().children('caption'));
+                    this._table().children('caption').removeClass('updating');
+                    this._loading = false;
+                },
+                (msg) => {
+                    console.log('request failed', this.fwk_app_name, msg);
+                    this._table().children('caption').html('<span style="color:maroon">No Response</span>');
+                    this._table().children('caption').removeClass('updating');
+                    this._loading = false;
+                }
+            );
+        }
+        
+        static arrayOfPairs2str(a) {
+            return _.reduce(a, function(s, pair){ return s + '<pre>' + pair[0] + ' (' + pair[1] + ')</pre> '; }, '');
+
+        }
+
+        _display(data) {
+            console.log(data);
+            let html = '';
+            for (let worker in data) {
+                if (!data[worker].success) {
+                    html += `
+<tr>
+  <th class="table-warning">${worker}</th>
+  <th class="table-secondary" colspan="6">&nbsp;</th>
+</tr>`;
+                } else {
+                    let schedulers = data[worker].info.processor.queries.blend_scheduler.schedulers;
+                    schedulers = _.sortBy(schedulers, 'priority');
+                    html += `
+<tr>
+  <th rowspan="${_.size(schedulers)+1}">${worker}</th>
+</tr>`;
```

Comment:
I'm not thrilled by the html indentation, but the fixes for that are about as painful as the original problem. For multiple nested loops, it may be worth making functions to return the strings, so it looks like
    html += _htmlSchedulerRow(schedulers, worker)

Reply:
Good suggestion! Thank you!